astropy/cosmology/core.py

```diff
@@ -863,20 +868,72 @@ def inv_efunc(self, z):
             Or = self._Ogamma0 + self._Onu0
         zp1 = 1.0 + z
 
-        return 1.0 / np.sqrt(zp1 ** 2 * ((Or * zp1 + Om0) * zp1 + Ok0) +
-                             Ode0 * self.de_density_scale(z))
+        return (zp1 ** 2 * ((Or * zp1 + Om0) * zp1 + Ok0) +
+                Ode0 * self.de_density_scale(z))**(-0.5)
+
+    # The idea is to eventually replace _tfunc with _tfunc_scalar,
+    #  switch public use over to tfunc.  The concern is that some people
+    #  may be using _tfunc publicly, despite the fact that it starts
+    #  with an underscore.
+    def _tfunc_scalar(self, z):
+        """ Integrand of the lookback time.
+
+        Parameters
+        ----------
+        z : float
+          Input redshift.
+
+        Returns
+        -------
+        I : float
+          The integrand for the lookback time
+
+        References
+        ----------
+        Eqn 30 from Hogg 1999.
+        """
+
+        args = self._scalar_inv_efunc_args
+        return self._scalar_inv_efunc(z, *args) / (1.0 + z)
 
+    @deprecated(since=1.1, alternative='tfunc',
+                message='will become scalar only')
```

Comment:
Ok. Either way is fine, as long as `_inv_efunc`, `_tfunc`, `_xfunc` follow the same naming convention.